aws/logs_monitoring/cache.py

```diff
@@ -435,10 +440,141 @@ def get(self, log_group):
             # If the custom tag fetch env var is not set to true do not fetch
             if not self.should_fetch_tags():
                 logger.debug(
-                    "Not fetching custom tags because the env variable DD_FETCH_LAMBDA_TAGS is not set to true"
+                    "Not fetching custom tags because the env variable DD_FETCH_LOG_GROUP_TAGS is "
+                    "not set to true"
                 )
                 return []
             log_group_tags = get_log_group_tags(log_group) or []
             self.tags_by_id[log_group] = log_group_tags
 
         return log_group_tags
+
+
+#######################
+# Step Functions Tags #
+#######################
+
+
+class StepFunctionsTagsCache(LambdaTagsCache):
+    CACHE_FILENAME = DD_S3_STEP_FUNCTIONS_CACHE_FILENAME
+    CACHE_LOCK_FILENAME = DD_S3_STEP_FUNCTIONS_CACHE_LOCK_FILENAME
+
+    def should_fetch_tags(self):
+        return should_fetch_step_functions_tags()
+
+    def build_tags_cache(self):
+        """Makes API calls to GetResources to get the live tags of the account's Step Functions
+        Returns an empty dict instead of fetching custom tags if the tag fetch env variable is not
+        set to true.
+        Returns:
+            tags_by_arn_cache (dict<str, str[]>): each Lambda's tags in a dict keyed by ARN
+        """
+        tags_fetch_success = False
+        tags_by_arn_cache = {}
+        get_resources_paginator = resource_tagging_client.get_paginator("get_resources")
+
+        try:
+            for page in get_resources_paginator.paginate(
```

Comment:
Just so I understand, we are following the lambda approach of prefetching the tags from each StepFunction, not the Cloudwatch Log group approach which doesn't fetch anything?

Reply:
Correct.